services/proxy-cache-service.ts

```diff
@@ -0,0 +1,39 @@
+import * as path from "path";
+import { Proxy } from "../constants";
+
+export class ProxyCacheService implements IProxyCacheService {
+	private proxyCacheFilePath: string;
+	private credentialsKey: string;
+
+	constructor(private $credentialsService: ICredentialsService,
+		private $fs: IFileSystem,
+		private $options: ICommonOptions,
+		private $staticConfig: Config.IStaticConfig) {
+		this.proxyCacheFilePath = path.join(this.$options.profileDir, Proxy.CACHE_FILE_NAME);
+		this.credentialsKey = `${this.$staticConfig.CLIENT_NAME}_PROXY`;
+	}
+
+	public setCache(cacheData: IProxyCache): IProxyCache {
+		this.$fs.writeJson(this.proxyCacheFilePath, cacheData);
+		return cacheData;
+	}
+
+	public getCache(): IProxyCache {
+		return this.$fs.exists(this.proxyCacheFilePath) && this.$fs.readJson(this.proxyCacheFilePath);
+	}
+
+	public clearCache(): void {
+		this.$fs.deleteFile(this.proxyCacheFilePath);
+		this.$credentialsService.clearCredentials(this.credentialsKey);
+	}
+
+	public async getCredentials(): Promise<ICredentials> {
+		return this.$credentialsService.getCredentials(this.credentialsKey);
```

Comment:
I think it is logically more appropriate that way, because if we have say `this.$staticConfig.PROXY_CREDENTIALS_KEY` that means that whoever want to get/set proxy credential details would have to know about `$staticConfig` as well as `$credentialsService`

Basically what I'm trying to say is - I think it's better to abstract all proxy-related things in one service instead of leaking logic to the code that intends to work with any proxy-related stuff.

I hope that makes sense